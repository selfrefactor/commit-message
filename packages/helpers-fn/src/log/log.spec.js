const { log } = require('./log')

let logSpy

beforeEach(() => {
  logSpy = jest.spyOn(global.console, 'log')
})
afterEach(() => {
  logSpy.mockRestore()
})

const str = 'FOO bar'

test('separator', () => {
  log('separator')
  // expect(logSpy.mock.calls[ 0 ]).toMatchSnapshot()
})

test('obj', () => {
  log({
    a : 1,
    b : [ 1, 2 ],
  }, 'obj')
  expect(logSpy.mock.calls[ 0 ]).toMatchSnapshot()
})

test('obj - when input is not an object', () => {
  log(str, 'obj')
  expect(logSpy.mock.calls[ 0 ]).toMatchSnapshot()
})

test('box', () => {
  log(str, 'box')
  expect(logSpy.mock.calls[ 0 ]).toMatchSnapshot()
})

test('with wrong mode', () => {
  const expectedSpyCalls = [
    'FOO bar',
    'not exist',
    'unrecognized helpers-fn.log mode',
  ]
  log(str, 'not exist')
  expect(logSpy.mock.calls[ 0 ]).toEqual(expectedSpyCalls)
})

test('with too many inputs', () => {
  const expectedSpyCalls = [
    1,
    2,
    3,
    'helpers-fn.log doesn\'t support multiple inputs',
  ]
  log(
    1, 2, 3
  )
  expect(logSpy.mock.calls[ 0 ]).toEqual(expectedSpyCalls)
})
