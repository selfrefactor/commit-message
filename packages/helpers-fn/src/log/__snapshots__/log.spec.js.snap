// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`box 1`] = `
Array [
  "┏━━━━━━━━━━━━━┓
┃             ┃
┃   FOO bar   ┃
┃             ┃
┗━━━━━━━━━━━━━┛",
]
`;

exports[`obj - when input is not an object 1`] = `
Array [
  "FOO bar",
]
`;

exports[`obj 1`] = `
Array [
  "{
  a: [36m1[39m,
  b: [
    [36m1[39m,
    [36m2[39m
  ]
}",
]
`;
